### DISCUSSION TITLE
Netlify won’t build my site using image tools

### DISCUSSION THREAD

**Discussion by rchrdnsh** [JonasKruckenberg/imagetools#105] [Q&A]
sooo, got this issue...figured out how to use cite-imagetools in a sveltekit app to process images defined in markdown and it works pretty darn well, taking my lighthouse performance score from the low 60's straight to 100, so awesome there!

Problem is that now netlify will not build my site anymore because it wants to use too much resources...ram, time, disk...not sure...so I ha e to build it on my machine then drag and drop the build folder into netlify to deploy. It works, but I would much prefer for that to not be the case.

I'm also noticing that every time I run `build` it processes the images again, even though they have already been processed many times.

So I'm wondering if there might be a way to process the images before they go to netlify, and also if there is a way to only process the images that change, rather than all of them every time.

**Comment by JonasKruckenberg** [comment]
Yeah this caching-feature was lost during the v2.x to v3.x rewrite, in favor of streaming images during dev mode. 
Having it broken in CI is obviously a no go, so I'll address this as soon as I can.

**Reply by rchrdnsh** [reply]
I'm noticing that it's the AVIF processing that's really doing it...if I switch to jpg then it might build, but when I switch to avif, then...nnnnnope! 🤣

Also streaming a bunch of avif files makes streaming...well...super slow...just a heads up...

**Reply by JonasKruckenberg** [reply]
Yeah the avif encoding itself is rather slow, this is out of my control (and probably a libvips) issue.
The caching feature is on my backlog though and will be done after the refactor!

**Comment by rchrdnsh** [comment]
Just curious if there has been any work in this area...thanks :-)

**Comment by alanbhamilton** [comment]
I'm bumping into this right now with a svelte-kit app building on Netlify. There are a couple hundred images and they are all being re-processed on every build, even though I'm caching the build folder using `netlify-plugin-cache`.

Is there a way of recognizing that the work has already been done and only processing images which haven't already been processed?

**Reply by benmccann** [reply]
Does Netlify have some limit for the length of time a build can run that you're bumping into?

**Reply by happycollision** [reply]
Even if a single build isn't failing for time limits, the accumulation of time does deplete monthly build minutes. (And just makes deployments much slower than they need to be.)
